Sources/AsyncDNSResolver/dnssd/DNSResolver_dnssd.swift

```diff
@@ -158,9 +162,32 @@ struct DNSSD {
                 return continuation.finish(throwing: AsyncDNSResolver.Error(dnssdCode: _code))
             }
 
+            let serviceSockFD = DNSServiceRefSockFD(serviceRefPtr.pointee)
+            guard serviceSockFD != -1 else {
+                return continuation.finish(throwing: AsyncDNSResolver.Error(code: .internalError))
+            }
+
+            var pollFDs = [pollfd(fd: serviceSockFD, events: Int16(POLLIN), revents: 0)]
+            while true {
+                guard !Task.isCancelled else {
+                    return continuation.finish(throwing: CancellationError())
+                }
+
+                let result = poll(&pollFDs, 1, 0)
+                guard result != -1 else {
+                    return continuation.finish(throwing: AsyncDNSResolver.Error(code: .internalError))
+                }
+
+                if result == 0 {
+                    continue
+                }
+                if result == 1 {
```

Comment:
According to `poll` man page, it's the number of elements in the `pollFDs` array that are ready for I/O. The array being of size 1, it could not be anything other than 1 if it is a positive number.

I can change it to `>=`, but that should never happen.